src/python/pants/reporting/streaming_workunit_handler.py

```diff
@@ -29,7 +35,7 @@ def end(self) -> None:
     # we report any workunits that were added after the last time the thread polled.
     workunits = self.scheduler.poll_workunits()
     for callback in self.callbacks:
-      callback(workunits)
+      callback(workunits=workunits, finished=True)
```

Comment:
might be late to the party here... however, how do we handle errors (Exceptions) raised from the callbacks ?
should we try to contain them ? do we need to add a `try-finally` clause to handle cleanup even if we don't want to catch exceptions ?
@Eric-Arellano @stuhood @gshuflin

Reply:
That's a good point - maybe we should wrap these calls in a catch all exceptions blog and log them with a warning? It makes sense that a broken plugin that registers itself to this callback shouldn't bring down the rest of pants.

Reply:
> It makes sense that a broken plugin that registers itself to this callback shouldn't bring down the rest of pants.

I'm less sure there... silently broken metrics gathering is pretty dangerous. Assuming that the plugin is not synchronously hitting the network right here (which we might expect to periodically fail), it's possible that broken plugins should break things here.